src/domain/core/src/entities/mod.rs

```diff
@@ -12,11 +12,13 @@ pub mod dataset;
 pub mod dataset_summary;
 pub mod engine;
 pub mod events;
+pub mod metadata_block_repository;
```

Comment:
At the time of your comment, we really shouldn't have stored `MetadataBlockRepository` trait in domain area.

But now, I had to add `MetadataChain::as_metadata_block_repository()` method that requires the trait to be in the area.

Reply:
I think `domain` is the right place for this trait, but can't it be in `domain/core/src/repos/` instead? It can still be imported by `MetadataChain` as it's within one crate.

Reply:
Done